### PR TITLE
Verbum: fix redirect & error

### LINKED ISSUE #9289: Social Menu & Social Media Icons: Add Strava Support
We received a request to add a [Strava](https://www.strava.com/) icon to both the Social Media Icons and the Social Menu.

Internal ref: 1081759-zen

### PR DESCRIPTION
**Pull request by arcangelini**
Fixes https://github.com/Automattic/jpop-issues/issues/9289

## Proposed changes:

* Reloads iframe after successful comment
* Adds `wp-cookie-consent` input to form
* Adjust `setcookie` to work from within iframe

What to be expected now

1. After a successful comment the form and page should refresh to add more comments.
2. When you comment completely anonymously you should not see your comment in the main list on the post if its not approved.
3. If you give your name and email and check the cookie consent you should see your comment awaiting moderation.

### Other information:

NA - Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

NA

## Does this pull request change what data or activity we track or use?

No

### Simple
1. Clean your sandbox and run `bin/jetpack-downloader test jetpack-mu-wpcom-plugin verbum/fix-stuff && bin/jetpack-downloader test jetpack verbum/fix-stuff`
4. Point your test site at your sandbox.
5. Go to your discussion settings and make sure all comments require moderation and that you can make anonymous comments.
6. Go to a post and test comments

### Jetpack Only
1. With your sandbox still setup from the Simple testing, point jetpack.wordpress.com to your sandbox.
2. Use the Jetpack Beta installer to check out the branch
4. Go to your discussion settings and make sure all comments require moderation and that you can make anonymous comments.
5. Go to a post and test comments

### Atomic
1. Follow same steps as Jetpack Only

### CODE REVIEW

**Review by heavyweight — APPROVED** [review]
Tested on simple, atomic and self hosted and looks good.
For the last two we would need to fix the redirect to the unmoderated link instead of back to the post, but that can be handled later.

File: projects/plugins/jetpack/modules/comments/base.php [hunk 1]
```diff
@@ -300,13 +300,37 @@ public function set_comment_cookies( $comment_id ) {
...
+
+			// Set samesite to None if the request is from Jetpack iframe.
+			// This is needed because it is considered third party.
+			if ( isset( $_REQUEST['for'] ) && 'jetpack' === $_REQUEST['for'] ) {
+				$cookie_options['samesite'] = 'None';
+			}
+			// phpcs:enable WordPress.Security.NonceVerification
+
+			// phpcs:disable Jetpack.Functions.SetCookie.MissingTrueHTTPOnly
+			isset( $comment->comment_author ) ? setcookie( 'comment_author_' . COOKIEHASH, $comment->comment_author, $cookie_options ) : null;
```
Comment (allilevine) [inline projects/plugins/jetpack/modules/comments/base.php:330]: Should these values from user input (e.g. `comment_author`) be validated/sanitized?

**Review by allilevine — COMMENTED** [review]
Atomic: My information is saved if I consent to the cookie. Whether I'm logged in, have the cookie, or don't, I see "comment sent", no refresh.

Screen Shot 2024-12-10 at 11 54 47 AM

Simple: My info is saved if I consent to the cookie. I see my comment waiting for moderation even if I don't have the cookie. Logged in commenting works.

Jetpack: My information is saved if I consent to the cookie. I see my comment waiting for moderation even if I don't have the cookie. There's no option to log in.

Screen Shot 2024-12-10 at 1 31 16 PM

Edit: forgot to note that the cookie is removed when I un-check consent, in all three.

File: projects/plugins/jetpack/modules/comments/base.php (hunk 1 above)
Comment (arcangelini) [reply projects/plugins/jetpack/modules/comments/base.php:330]: Great question, this function is running after the comment has been fully processed and already stored in the database. These values that we are getting have already been validated and sanitized up stream. :thumbsup:

**Pull request merged**